third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -663,30 +663,37 @@ impl Interpreter {
                 let arena_id = traversal_context
                     .referenced_module_ids
                     .alloc(module_name.clone());
-                resolver
-                    .loader()
-                    .check_dependencies_and_charge_gas(
-                        module_store,
-                        data_store,
-                        gas_meter,
-                        &mut traversal_context.visited,
-                        traversal_context.referenced_modules,
-                        [(arena_id.address(), arena_id.name())],
-                    )
-                    .map_err(|err| {
-                        PartialVMError::new(err.major_status())
-                            .with_message(format!("Module {} failed to be charged", module_name))
-                    })?;
-                resolver
-                    .loader()
-                    .load_module(&module_name, data_store, module_store)
-                    .map_err(|_| {
-                        PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
-                            .with_message(format!("Module {} doesn't exist", module_name))
-                    })?;
+                if let Ok(true) = data_store.exists_module(&module_name) {
```

Comment:
This call bypasses all caches, an issues a read to storage, loading all bytes. I wish we had on-demand loading so that we do not have to do this and just `load_module()?`. Cannot we do so already? Or the error message there does not work?

Reply:
The problem was that `check_dependencies_and_charge_gas` will throw really weird error when the module doesn't exist. Do you have a better idea how to do this properly? cc @vgao1996

Reply:
What is the error? I thought it is "Linker Error: Cannot find {:?} in data cache"?

Reply:
The error msg looks like: `LINKER_ERROR (code 1081) error with error message: "Module 000000000000000000000000000000000000000000000000000000000000face::deflation_token failed to be charged".` and it's a bit hard to read.

Reply:
Can we remove this error remapping all together? Seems like the error message should originate from the inside of `check_dependencies_and_charge_gas`?

Reply:
Updated the error msg! Now the error will look like this:
```
Linker Error: Module 000000000000000000000000000000000000000000000000000000000000face::deflation_token doesn't exist.Failed to charge transitive dependency for 000000000000000000000000000000000000000000000000000000000000face::deflation_token. Does this module exists?
```